pkg/api/v1alpha1/authenticated_user.go

```diff
@@ -520,6 +527,27 @@ func (r *Router) updateAuthenticatedUser(c *gin.Context) {
 		return
 	}
 
+	updateNotificationPublishEventErr := error(nil)
+
+	_, status, err := handleUpdateNotificationPreferencesRequests(
+		c, tx, ctxUser, r.EventBus, req.NotificationPreferences,
+	)
+	if err != nil && !errors.Is(err, ErrNotificationPreferencesEmptyInput) {
+		if errors.Is(err, ErrPublishUpdateNotificationPreferences) {
```

Comment:
just a thought - wouldn't it be simpler to duplicate some of the code for updating notification preferences here and the other places where we're trying to use abstracted code? 😄  i believe other cases are handled that way.

Reply:
I was doing that, until I had to do the same update after I changed the model in four different places, then I decided it should be abstracted 😄